FileSystem/file_system.js

```diff
@@ -37,3 +37,101 @@ FileSystem.getFolder = function(localPath) {
   localPath = DirectoryManager.getAbsolutePath(localPath);
   return new VbaFolder(localPath);
 };
+
+/**
+ * Emulates VBA FileSystemObject.BuildPath API
+ * Combines a folder path and the name of a folder or file and returns the
+ * combination with valid path separators. Method doesn't check if the file or
+ * folder actually exists.
+ * @param {string} localPath Local file path of the folder. Can be either
+ *     relative or absolute path.
+ * @param {string} name Existing path with which name is combined.
+ * @return {string} Combined file and folder path
+ */
+FileSystem.buildPath = function(localPath, name) {
+  var lastChar = localPath[localPath.length - 1];
+  if (lastChar != '/' && lastChar != '\\') {
+    localPath += '\\';
+  }
+  return localPath + name;
+};
+
+/**
+ * Emulates VBA FileSystemObject.GetTempName API
+ * Get a randomly generated temporary file or folder name that is useful for
+ * performing operations that require a temporary file or folder.
+ * @return {string} Temporary file or folder name
+ */
+FileSystem.getTempName = function() {
+  var prefix = 'rad';
+  var suffix = '.tmp';
+  var randomValues = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
+  var randomPart = '';
+  for (var i = 0; i < 5; i++) {
+    var index = Math.floor(Math.random() * randomValues.length);
+    randomPart += randomValues[index];
+  }
+  return prefix + randomPart + suffix;
+};
+
+/**
+ * Emulates VBA FileSystemObject.GetExtensionName
+ * Get a string containing the extension name for the last component in a path.
+ * @param {string} localPath Local file path of the file
+ * @returns {string} The extension name
+ */
+FileSystem.getExtensionName = function(localPath) {
+  var fileExtensionRegExp = /\.(\w*)$/;
```

Comment:
will this match abc.txt2 ?

Reply:
![image](https://user-images.githubusercontent.com/66052573/87708302-22388880-c7c0-11ea-8ac8-b55ac056ed03.png)

Yes, it matches `abc.txt2`.